agent/hibinit-agent

```diff
@@ -353,55 +302,53 @@ def identify_file_system(swapfile, file_system):
 
 
 class Config(object):
-    def __init__(self, config, args):
-        def get(section, name):
+    def __init__(self, config_file, args):
+        def get_from_config(section, name):
             try:
-                return config.get(section, name)
+                return config_file.get(section, name)
             except NoSectionError:
                 return None
             except NoOptionError:
                 return None
 
-        def get_int(section, name):
-            v = get(section, name)
-            if v is None:
-                return None
-            return int(v)
+        def get_int_from_config(section, name):
+            v = get_from_config(section, name)
+            return None if v is None else int(v)
 
         self.log_to_syslog = self.merge(
-            self.to_bool(get('core', 'log-to-syslog')), self.to_bool(args.log_to_syslog), True)
+            self.to_bool(args.log_to_syslog),
+            self.to_bool(get_from_config('core', 'log-to-syslog')),
+            True)
+        self.grub_update = self.merge(
+            self.to_bool(args.grub_update),
+            self.to_bool(get_from_config('core', 'grub-update')),
+            True)
 
-        self.mkswap = self.merge(get('swap', 'mkswap'), args.mkswap, 'mkswap {swapfile}')
-        self.swapon = self.merge(get('swap', 'swapon'), args.swapon, 'swapon {swapfile}')
-        self.swapoff = self.merge(get('swap', 'swapoff'), args.swapoff, 'swapoff {swapfile}')
         self.touch_swap = self.merge(
-            self.to_bool(get('core', 'touch-swap')), self.to_bool(args.touch_swap),
+            self.to_bool(args.touch_swap),
+            self.to_bool(get_from_config('core', 'touch-swap')),
             identify_file_system(SWAP_FILE, "xfs"))
         self.btrfs_enabled = self.merge(
-            self.to_bool(get('core', 'btrfs-enabled')), self.to_bool(args.btrfs_enabled),
+            self.to_bool(args.btrfs_enabled),
+            self.to_bool(get_from_config('core', 'btrfs-enabled')),
             identify_file_system(SWAP_FILE, "btrfs"))
+        self.state_dir = self.merge(None, get_from_config('core', 'state-dir'), DEFAULT_STATE_DIR)
 
-        self.grub_update = self.merge(
-            self.to_bool(get('core', 'grub-update')), self.to_bool(args.grub_update), True)
-
-        self.swap_percentage = self.merge(
-            get_int('swap', 'percentage-of-ram'), args.swap_ram_percentage, 100)
-        self.swap_mb = self.merge(
-            get_int('swap', 'target-size-mb'), args.swap_target_size_mb, 4000)
+        self.swap_percentage = self.merge(args.swap_ram_percentage, get_int_from_config('swap', 'percentage-of-ram'), 100)
+        self.swap_mb = self.merge(args.swap_target_size_mb, get_int_from_config('swap', 'target-size-mb'), 4000)
```

Comment:
is 4000 the default value ?  if yes may be creating separate constant is better

Reply:
Yes, missed this magic number, will create a constant instead.